integration/test_collection.py

```diff
@@ -1445,7 +1445,14 @@ def test_iterator(client: weaviate.Client, count: int):
         assert all_data == list(range(count))
```

Comment:
we should make sure it works for all combinations of
prop: None and not None
metadata None and not None

Note you can add two paramaterize on top of each other, eg
```
@pytest.mark.parametrize("props",...)
@pytest.mark.parametrize("metadata",...)
def ...
```

and it will do all combinations of the two